src/main/java/edu/harvard/iq/dataverse/dataaccess/S3AccessIO.java

```diff
@@ -0,0 +1,593 @@
+import org.apache.commons.io.IOUtils;
+
+
+/**
+ *
+ * @author Matthew A Dunlap
+ * @author Sarah Ferry
+ * @author Rohit B
+ * @author Brian Silverstein
+ * @param <T> what it stores
+ */
+/* 
+    Experimental Amazon AWS S3 driver
+ */
+public class S3AccessIO<T extends DvObject> extends StorageIO<T> {
+
+    private static final Logger logger = Logger.getLogger("edu.harvard.iq.dataverse.dataaccess.S3AccessIO");
+
+    public S3AccessIO() {
+        this(null);
+    }
+
+    public S3AccessIO(T dvObject) {
+        this(dvObject, null);
+    }
+    
+    public S3AccessIO(T dvObject, DataAccessRequest req) {
+        super(dvObject, req);
+        this.setIsLocalFile(false);
+        awsCredentials = new ProfileCredentialsProvider().getCredentials();
+        s3 = AmazonS3ClientBuilder.standard().withCredentials(new AWSStaticCredentialsProvider(awsCredentials)).withRegion(Regions.US_EAST_1).build();
+        
+    }
+
+    private AWSCredentials awsCredentials = null;
+    private AmazonS3 s3 = null;
+    private String bucketName = System.getProperty("dataverse.files.s3-bucket-name");
+    private String s3FolderPath;
+    private String s3FileName;
+    private String storageIdentifier;
+  
+    @Override
+    public void open(DataAccessOption... options) throws IOException {
+        if(s3==null)
+        {
+            throw new IOException("ERROR: s3 not initialised. ");
+        }
+        
+        if(bucketName == null || !s3.doesBucketExist(bucketName)) { 
+            throw new IOException("ERROR: S3AccessIO - You must create and configure a bucket before creating datasets.");
+        } 
+        
+        DataAccessRequest req = this.getRequest();
+        S3Object s3object=null;
+        
+        if (isWriteAccessRequested(options)) {
+            isWriteAccess = true;
+            isReadAccess = false;
+        } else {
+            isWriteAccess = false;
+            isReadAccess = true;
+        }
+        
+//        Dataset dataset = (Dataset)dvObject;
+        
+        //FIXME: Finish? 
+        if (dvObject instanceof DataFile) {
+            
+            DataFile dataFile = this.getDataFile();
+            s3FolderPath = this.getDataFile().getOwner().getAuthority() + "/" + this.getDataFile().getOwner().getIdentifier();
+//            s3FileName = this.getDataFile().getStorageIdentifier();
+            
+            if (req != null && req.getParameter("noVarHeader") != null) {
+                this.setNoVarHeader(true);
+            }
+            
+            if (dataFile.getStorageIdentifier() == null || "".equals(dataFile.getStorageIdentifier())) {
+                throw new FileNotFoundException("Data Access: No local storage identifier defined for this datafile.");
+            }
+            if (isReadAccess) {
+                storageIdentifier = dvObject.getStorageIdentifier();
+                if (storageIdentifier.startsWith("s3://")) {
+                    logger.info("Rohit BHATTA: "+storageIdentifier);
```

Comment:
Please remove developer's names from logging errors. Also, this should be `fine` rather than `info`.